### PR TITLE
vault-dev-autounseal - proper vault auto unseal golang automation

### PR DESCRIPTION
**Pull request by Alevsk**

### CODE REVIEW

File: cluster/kes.go [hunk 1]
```diff
@@ -426,8 +460,9 @@ func storeKeyPairInSecret(secretName string, content map[string]string) <-chan s
 	return doneCh
 }
 
-func generateKeyPairAndStoreInSecret(name string) *KeyPair {
-	kesKeyPair := <-generateKeyPair(name)
+func generateKeyPairAndStoreInSecret(name string, tenant string) *KeyPair {
```
Comment (dvaldivia) [inline cluster/kes.go:466]: Add a comment about how to function works

**Review by dvaldivia — DISMISSED** [review]

File: vault/go.mod [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+require (
+	github.com/fatih/color v1.7.0
+	github.com/hashicorp/vault/api v1.0.5-0.20191216174727-9d51b36f3ae4
+	github.com/minio/minio v0.0.0-20191230055646-8eba97da74ef
+)
+
+replace github.com/gorilla/rpc v1.2.0+incompatible => github.com/gorilla/rpc v1.2.0
+
+replace github.com/hashicorp/vault => github.com/hashicorp/vault v1.3.1
```
Comment (harshavardhana) [inline vault/go.mod]: You don't need to do this anymore @Alevsk please update the dependencies

**Review by harshavardhana — DISMISSED** [review]

**Review by dvaldivia — APPROVED** [review]

**Comment by harshavardhana** [comment]
@aead can you look at this PR if vault related policies are correct?

**Comment by dvaldivia** [comment]
LGTM, I'm going to merge it to master. @aead if there's comments or concerts @Alevsk can address them on a separate PR

**Pull request merged**